ofono2mm/mm_modem.py

```diff
@@ -120,13 +128,122 @@ async def init_mm_3gpp_interface(self):
         self.bus.export('/org/freedesktop/ModemManager1/Modem/' + str(self.index), self.mm_modem3gpp_interface)
         self.mm_modem3gpp_interface.set_props()
 
+    async def init_mm_simple_interface(self):
+        self.mm_modem_simple_interface = MMModemSimpleInterface(self)
+        self.bus.export('/org/freedesktop/ModemManager1/Modem/' + str(self.index), self.mm_modem_simple_interface)
+
     async def init_mm_messaging_interface(self):
         self.mm_modem_messaging_interface = MMModemMessagingInterface(self.index, self.bus, self.ofono_proxy, self.modem_name, self.ofono_modem, self.ofono_props, self.ofono_interfaces, self.ofono_interface_props)
         self.bus.export('/org/freedesktop/ModemManager1/Modem/' + str(self.index), self.mm_modem_messaging_interface)
         if 'org.ofono.MessageManager' in self.ofono_interfaces:
             self.mm_modem_messaging_interface.set_props()
             await self.mm_modem_messaging_interface.init_messages()
 
+    async def check_ofono_contexts(self):
+        global bearer_i
+        if not 'org.ofono.ConnectionManager' in self.ofono_interfaces:
+            return
+        contexts = await self.ofono_interfaces['org.ofono.ConnectionManager'].call_get_contexts();
+        old_bearer_list = self.props['Bearers'].value
+        for ctx in contexts:
+            if ctx[1]['Type'].value == "internet":
+                mm_bearer_interface = MMBearerInterface(self.index, self.bus, self.ofono_proxy, self.modem_name, self.ofono_modem, self.ofono_props, self.ofono_interfaces, self.ofono_interface_props, self)
+                ip_method = 0
+                if 'Method' in ctx[1]['Settings'].value:
+                    if ctx[1]['Settings'].value['Method'].value == "static":
+                        ip_method = 2
+                    if ctx[1]['Settings'].value['Method'].value == "dhcp":
+                        ip_method = 3
+                ip_address = ''
+                if 'Address' in ctx[1]['Settings'].value:
+                    ip_address = ctx[1]['Settings'].value['Address'].value
+                ip_dns = []
+                if 'DomainNameServers' in ctx[1]['Settings'].value:
+                    ip_dns = ctx[1]['Settings'].value['DomainNameServers'].value
+                ip_gateway = ''
+                if 'Gateway' in ctx[1]['Settings'].value:
+                    ip_gateway = ctx[1]['Settings'].value['Gateway'].value
+                mm_bearer_interface.props.update({
+                    "Interface": ctx[1]['Settings'].value['Interface'] if 'Interface' in ctx[1]['Settings'].value else Variant('s', ''),
```

Comment:
`ctx[1]['Settings'].value.get("Interface", Variant('s', ''))` should work as well but keeps the line shorter